akvo/rsr/views/py_reports/kickstart_word_report.py

```diff
@@ -108,11 +116,231 @@ def render_report(request, project_id):
+
+    legend_table.rows[3].cells[0].paragraphs[-1].add_run().add_picture(load_image(IMG_GRADE_LOW))
+    legend_table.rows[3].cells[1].text = '0% - 49%'
+    legend_table.rows[3].cells[2].text = 'Result under reached'
+
+    doc.add_paragraph('')
+
+    quantitative_table = doc.add_table(rows=1, cols=3)
+    quantitative_table.cell(0, 0).merge(quantitative_table.cell(0, 2))
+    title_p = set_repeat_table_header(quantitative_table.rows[0]).cells[0].paragraphs[-1]
+    title_p.text = 'Results and indicators (quantitative)'
+    title_p.style = 'Heading 2'
+    for result in project_view.results:
+        if not result.has_quantitative_indicators:
+            continue
+        row = quantitative_table.add_row()
+        row.cells[0].merge(row.cells[2])
+        result_title_p = row.cells[0].paragraphs[-1]
+        result_title_p.text = result.title
+        result_title_p.style = 'Heading 3'
+        for indicator in result.indicators:
+            if not indicator.is_quantitative:
+                continue
+            row = quantitative_table.add_row()
+            row.cells[0].text = indicator.title
+            row.cells[1].text = indicator.progress_str
+            grade_image = IMG_GRADE_HIGH if indicator.grade == 'high' \
+                else IMG_GRADE_MEDIUM if indicator.grade == 'medium' \
+                else IMG_GRADE_LOW
+            row.cells[2].paragraphs[-1].add_run().add_picture(load_image(grade_image))
+
+    doc.add_paragraph('')
+
+    doc.add_heading('Results and indicators (qualitative)', 2)
+    for result in project_view.results:
+        if not result.has_qualitative_indicators:
+            continue
+        for indicator in result.indicators:
+            if not indicator.is_qualitative:
+                continue
+            doc.add_heading(indicator.title, 3)
+            for period in indicator.periods:
+                if not period.has_qualitative_data:
+                    continue
+                doc.add_paragraph('{} to {}'.format(
+                    period.period_start.strftime('%Y-%m-%d'), period.period_end.strftime('%Y-%m-%d')))
+                for update in period.approved_updates:
+                    markdown_to_docx(doc.add_paragraph(), update.narrative)
+                    if update.photo:
+                        doc.add_paragraph().add_run().add_picture(load_image(update.photo_url), width=Mm(100))
+                    if update.file:
+                        add_hyperlink(doc.add_paragraph(), update.file_url, update.file_url)
+                    doc.add_paragraph('\n\n')
+
+    doc.add_heading('Project updates', 1)
+    for project_update in project_updates:
+        doc.add_heading(project_update.title, 2)
+        doc.add_paragraph(project_update.created_at.strftime('%Y-%m-%d'))
+        doc.add_paragraph().add_run().add_picture(load_image(project_update.photo_url), width=Mm(100))
+        markdown_to_docx(doc.add_paragraph(), project_update.text)
```

Comment:
This seems to crash if the update has no text
I think the `markdown_to_docx` function should check for empty text and handle it?

Reply:
Should be fixed with the new commits.